gym/spaces/box.py

```diff
@@ -25,6 +25,9 @@ def _short_repr(arr: np.ndarray) -> str:
     return str(arr)
 
 
+BOX_SHAPE_TYPE = Tuple[Union[int, "BOX_SHAPE_TYPE"], ...]
```

Comment:
Why do we need a recursive type? I love them, but (a) not in python, and (b) I don't think we allow shapes like (2, (3, 4), 5)

Reply:
I was getting confused with `MultiDiscrete` where the shape is recursive and weird. Good spot

Reply:
I have cast all of the shape dims to integers so there shouldn't be any need to the asserts anymore